sfcc_cartridge/int_fast_sfra/cartridge/static/default/js/fastProduct.js

```diff
@@ -11,26 +11,39 @@ $(document).ready(function () {
         checkoutFastButton.removeClass('fast-checkout');
     }
 
+    // following code will exucute when click on quickview button which having class 'quickview'
+    $('body').on('click', '.quickview', function (e) {
+        e.preventDefault();
+        
+        setTimeout(function() {
+            initQuickviewFastBtn();
+        }, 1000);
+    });
+
     $('body').on('product:updateAddToCart', function (e, response) {
+        var isQuickViewModalOpen = document.querySelector('#quickViewModal') != null ? document.querySelector('#quickViewModal').classList.contains('show') : false;
+        if (isQuickViewModalOpen) {
+            checkoutFastButton = $('#fastQuickviewCheckoutButton');
+        }
+        checkoutFastButton.attr('disabled', (!response.product.readyToOrder || !response.product.available));
+        checkoutFastButton.attr('product_id', response.product.id);
+        addEventToFastButton(response);
+    });
+    
+    function initQuickviewFastBtn() {
         var isQuickViewModalOpen = document.querySelector('#quickViewModal') != null ? document.querySelector('#quickViewModal').classList.contains('show') : false,
-            isEditProductModalOpen = document.querySelector('#editProductModal') != null ? document.querySelector('#editProductModal').classList.contains('show') : false,
             quickviewQty = 1;
             
-        if (isQuickViewModalOpen || isEditProductModalOpen) {
+        if (isQuickViewModalOpen) {
             checkoutFastButton = $('#fastQuickviewCheckoutButton');
             
             if ( checkoutFastButton !== null && checkoutFastButton.length > 0) {
                 
                 checkoutFastButton.on('click', function() {
                     if (isQuickViewModalOpen) {
-                    	quickviewQty = parseInt( $('#quickViewModal .quantity-select').find(":selected").val().trim() );
-                    }
-                    
-                    if (isEditProductModalOpen) {
-                    	quickviewQty = parseInt( $('#editProductModal .quantity-select').find(":selected").val().trim() );
+                        quickviewQty = parseInt( $('#quickViewModal .quantity-select').find(":selected").val().trim(), 10 );
```

Comment:
What if `$('#quickViewModal .quantity-select')` not found - this will throw exception . should we check if found and selected before getting val and trim?

Reply:
@igorzinc every quickview popup having quantity select.
so this scenario will not come